app/src/main/java/at/aau/edu/wizards/MainActivity.kt

```diff
@@ -43,15 +49,50 @@ class MainActivity : AppCompatActivity() {
     }
 
     private fun setupUi() {
+        val sharedPrefs: SharedPreferences = getSharedPreferences(getString(R.string.shared_storage), Context.MODE_PRIVATE)
```

Comment:
do we need SharedPreferences? I'm new to this aswell, but as far as i have red this uses a lot of resources and is used to communicate between activities. We only use one activity though, so is there a reason you decided to implement this instead of just saving it locally inside the activity?

Reply:
Yes, but as far as I can see if the application as a whole gets killed we still loose the data and since there is only one activity in our application there won't be a case where we it will be killed. But I guess we just future proof with it?

Reply:
Activity can be killed any time, you can get a phone call and the systems kills our Activity because low memory. But we handle such restoration cases not properly anyhow. Here it's more about avoiding entering name when opening the app second (third,...) time

Reply:
as discussed on DD,  we stick to using SharedPrefs until we find a better solution